lib/features/api_interact_example/presentation/screens/api_interact_example/api_interact_example_model.dart

```diff
@@ -0,0 +1,43 @@
+import 'package:elementary/elementary.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter_template/core/architecture/domain/entity/result.dart';
+import 'package:flutter_template/features/api_interact_example/domain/repositories/i_api_interact_example_repository.dart';
+import 'package:flutter_template/features/api_interact_example/presentation/screens/api_interact_example/api_interact_example_screen.dart';
+import 'package:union_state/union_state.dart';
+
+/// Typedef for [UnionState] IP address.
+typedef IpUnionState = UnionState<String?>;
+
+/// {@template api_interact_example_model.class}
+/// [ElementaryModel] for [ApiInteractExampleScreen]
+/// {@endtemplate}
+final class ApiInteractExampleModel extends ElementaryModel {
+  final IApiInteractExampleRepository _repository;
+
+  /// {@macro api_interact_example_model.class}
+  ApiInteractExampleModel({
+    required IApiInteractExampleRepository repository,
+    super.errorHandler,
+  }) : _repository = repository;
+
+  /// IP address state.
+  final _ipState = UnionStateNotifier<String?>(null);
```

Comment:
Let's not use UnionStateNotifier. And just make `ApiInteractExampleState` sealed class which will contain initial, loading, loaded and error states.